src/main/java/edu/oregonstate/cs361/battleship/Hard.java

```diff
@@ -1,13 +1,93 @@
 package edu.oregonstate.cs361.battleship;
 
+import java.util.Random;
+
 /**
  * Created by Keana on 3/15/2017.
  */
 public class Hard extends Mode {
 
+    boolean hit;
+    int count;//tracks directions checked around a hit
+    Coordinate lastHit;
+    Coordinate hitPrime;
+
+
     public Hard(){
 
     }
+    private void _fire(BattleshipModel bs, Coordinate shot){//helper function for fire

```

Comment:
This function cannot work with the way the program is currently implemented. The battleship model creates a mode class and cannot pass itself to this child class.